src/components/container/ConnectDevicesMenu/ConnectDevicesMenu.tsx

```diff
@@ -49,7 +53,8 @@ function useConnectDevicesMenuState(
         deviceExternalAccounts: null,
         platform: null,
         hasRecentAppleHealthData: false,
-        healthConnectStatus: null
+        healthConnectStatus: null,
+        participantInfo: null
     });
 
     if (previewState) {
```

Comment:
Pre-existing condition, but I think this whole preview block should be moved to be the first thing inside the `initialize()` function, and it should set the `state` to be the preview state rather than returning it.  As-is, this technically breaks the React Rules Of Hooks, because the `useInitializeView(...)` hook call below (which calls `useEffect` internally) is being called conditionally.

It is splitting hairs a little bit because we are not likely to swap between a preview state and a non-preview state during a particular use of the component, but that possibility does exist.

Reply:
I'd prefer not to make a change like that here in a (mostly) unrelated PR.  I generally agree with you because when I came to edit this component I was kind of confused by the order of events, and it uses a different pattern than we've generally established (`useState` gets called in `useConnectDevicesMenuState` and then gets decomposed in the main component).  But I'd rather do that and test it in its own branch/PR as a bug fix/optimization.